### PR TITLE
Handle resource management (services, storage)

### LINKED ISSUE #851: One sees the logs of the other connected users
I can see the logs of the sidecar from @mguidon while connected as @sanderegg 

need to show only relevant logs.

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

- servicelib:
  - adds observer pattern in webserver, by using python decorators
- webserver:
  - added socketio subsystem in webserver:
    - handles websocket connection in webserver by authenticating user and creating a registry of user_id - socket_id using the Redis service
    NOTE: Currently disallow connecting to the websocket without passing a unique id (which identify a client session id or the tab)
    - actively disconnects all user websockets when user logs out in one tab
    - removed dependency to director_sdk in webserver and removed director-sdk package

  - added resource_manager subsystem in webserver:
    - handle delayed closing of projects (thus deletion of services) when the user disconnects (using ENV WEBSERVER_SERVICES_DELETION_TIMEOUT_SECONDS)
    - handles disconnection/reconnection of websockets without logging out
    - fixes several >1 tabs use cases (close in one tab while project open in other tab, prevent deleting a project opened in another tab or user)
    - deleting a project now remove all running interactive services (fire&forget)
- director:
  - director now allows for getting a list of running interactive services with an optional filter (user_id and/or project_id)
  - updated openapi-generator-cli to v4.2.1 to generate director API handlers
  

- Frontend (thanks to @odeimaiz):
  - Watchdog relies on websocket connection instead of calling the healthcheck rest API
  - Keep the studyId to be loaded in osparc.store.Store instead of passing it as argument through all constructors
  - Show a circular icon instead of the tick when inputs retrieval is successful but data downloaded is 0
  - InteractiveServices Calls moved to Resources
  - Allow endPoint specific useCache flag
  - When reloading site, load the activeStudy if any
  - Add client_session_id to websocket query

bonus:
- enable tracing through aiozipkin library (server side) in each core service providing a REST api
- only send logs/progress to websocket client identified by the user_id fixes #851 
- make use of [watchdog](https://github.com/gorakhargosh/watchdog) to auto-restart python process on file change
- adds Redis-Commander to visualize the redis db in devel mode
- re-organized simcore-sdk package unit/integration tests to follow same procedure as webserver tests

## How to test

```
make build-devel
# this starts with a service deletion delay of 15 seconds
make up-devel
```
or
```
make build
# change the service deletion delay from 15 minutes to something smaller 
make up
```

Current defaults:
- Garbage collector runs every 30 seconds (see  WEBSERVER_GARBAGE_COLLECTION_INTERVAL_SECONDS in .env-devel)
- Service deletion delay is 15 minutes in production mode, 15 seconds in devel mode (see WEBSERVER_RESOURCES_DELETION_TIMEOUT_SECONDS in .env-devel)

## Checklist

- [ ] I think the code is well written
- [x] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [ ] **Runs in the swarm**
- [ ] If you design a new module, add your user to .github/CODEOWNERS

### CODE REVIEW

**Comment by ignapas** [comment]
holy cow, that is a big PR right there

File: services/web/server/src/simcore_service_webserver/computation_subscribe.py [hunk 1]
```diff
@@ -1,28 +1,50 @@
...
-        if data["Channel"] == "Log":
-            await sio.emit("logger", data = json.dumps(data))
-        elif data["Channel"] == "Progress":
-            await sio.emit("progress", data = json.dumps(data))
-        asyncio.sleep(2)
+        user_id = data["user_id"]
+        with managed_resource(user_id, None, app) as rt:
+            socket_ids = await rt.find_socket_ids()
+            for sid in socket_ids:
+                # we only send the data to the right sockets
```
Comment (mguidon) [inline services/web/server/src/simcore_service_webserver/computation_subscribe.py]: UNBELIEVABLE!!

**Comment by sanderegg** [comment]
> Is the expected behavior/lifetime of services explained in some document? It would be nice to know how to test this. For the moment I have created a couple of studies and every service stays up... going back to the dashboard and closing tabs do not close them.

@ignapas :
I added the default in the PR description. Basically the service deletion delay is set by default in production mode to 15 minutes. 15 seconds in devel mode.

**Comment by ignapas** [comment]
> > Is the expected behavior/lifetime of services explained in some document? It would be nice to know how to test this. For the moment I have created a couple of studies and every service stays up... going back to the dashboard and closing tabs do not close them.
> 
> @ignapas :
> I added the default in the PR description. Basically the service deletion delay is set by default in production mode to 15 minutes. 15 seconds in devel mode.
> e.g. going back to the dashboard or closing tab should take approx 15 minutes before they are switched off if you did not change the envs in webserver.

Exactly... I am always running ``make up-devel``... after closing all tabs and going for a coffee, they are still there. Maybe I am missing something!

**Comment by sanderegg** [comment]
> > > Is the expected behavior/lifetime of services explained in some document? It would be nice to know how to test this. For the moment I have created a couple of studies and every service stays up... going back to the dashboard and closing tabs do not close them.
> > 
> > 
> > @ignapas :
> > I added the default in the PR description. Basically the service deletion delay is set by default in production mode to 15 minutes. 15 seconds in devel mode.
> > e.g. going back to the dashboard or closing tab should take approx 15 minutes before they are switched off if you did not change the envs in webserver.
> 
> Exactly... I am always running `make up-devel`... after closing all tabs and going for a coffee, they are still there. Maybe I am missing something!

hmm, no seems not good..let me have a look... maybe something is screwed.

**Comment by sanderegg** [comment]
> > > > Is the expected behavior/lifetime of services explained in some document? It would be nice to know how to test this. For the moment I have created a couple of studies and every service stays up... going back to the dashboard and closing tabs do not close them.
> > > 
> > > 
> > > @ignapas :
> > > I added the default in the PR description. Basically the service deletion delay is set by default in production mode to 15 minutes. 15 seconds in devel mode.
> > > e.g. going back to the dashboard or closing tab should take approx 15 minutes before they are switched off if you did not change the envs in webserver.
> > 
> > 
> > Exactly... I am always running `make up-devel`... after closing all tabs and going for a coffee, they are still there. Maybe I am missing something!
> 
> hmm, no seems not good..let me have a look... maybe something is screwed.

ok it's fixed. bad mock.

**Pull request merged**